internal/controllers/device_data_controller.go

```diff
@@ -398,15 +400,62 @@ func (udc *UserDevicesController) GetUserDeviceErrorCodeQueries(c *fiber.Ctx) er
 			return err
 		}
 
-		queries = append(queries, GetUserDeviceErrorCodeQueriesResponseItem{
+		userDeviceresp := GetUserDeviceErrorCodeQueriesResponseItem{
 			ErrorCodes:  ercJSON,
 			RequestedAt: erc.CreatedAt,
-		})
+		}
+
+		/* if !erc.ClearedAt.Valid {
+			userDeviceresp.ClearedAt =
+		} */
+
+		queries = append(queries, userDeviceresp)
 	}
 
 	return c.JSON(GetUserDeviceErrorCodeQueriesResponse{Queries: queries})
 }
 
+// ClearUserDeviceErrorCodeQuery godoc
+// @Description Clear user device error codes
+// @Tags        user-devices
+// @Success     200 {object} controllers.QueryDeviceErrorCodesResponse
+// @Security    BearerAuth
+// @Router      /user/devices/{userDeviceID}/error-codes/clear [post]
+func (udc *UserDevicesController) ClearUserDeviceErrorCodeQuery(c *fiber.Ctx) error {
+	udi := c.Params("userDeviceID")
+
+	logger := helpers.GetLogger(c, udc.log)
+
+	errCodeQuery, err := models.ErrorCodeQueries(
+		models.ErrorCodeQueryWhere.UserDeviceID.EQ(udi),
+		qm.OrderBy(models.ErrorCodeQueryColumns.CreatedAt+" DESC"),
+		qm.Limit(1),
+	).One(c.Context(), udc.DBS().Reader)
+	if err != nil {
+		logger.Err(err).Msg("error occurred when fetching error codes for device")
+		return fiber.NewError(fiber.StatusInternalServerError, "error occurred fetching device error queries")
```

Comment:
Can we throw a 4xx if there are no queries? I just like to avoid 5xx if it's something expected.